[Projects Page] initial work on setting up a projects page for tracking progress (#35)
executablebooks/meta: docs/projects.md

# Projects

The following is a list of projects that this project will support and develop. The following projects fit together to form the 
Executable Book Project. 

```{contents} Contents:
```

## Overview

{{ add Version 3 build diagram }}

High level feature lists are provided as a simple measure of progress. `Phase 1` are items required for minimal viable product, `Phase 2` include items that are planned for initial release, and 
`Phase 3` captures wishlist items to be implemented in the future.

## Authoring Tools

> ### [JupyterText](https://github.com/mwouts/jupytext) [External]

Reviewer: ```suggestion
### [Jupytext](https://github.com/mwouts/jupytext) [External]
```

**Description:** 

Jupyter Notebooks as Markdown Documents, Julia, Python or R scripts

**Contribution:** 

The primary aim is to add myst to jupytext to enable *bi-directional* conversion 
between IPYNB and myst text based files. 

> A seconday objective is to understand how JupyText may be used to convert between 

Reviewer: ```suggestion
A secondary objective is to understand how JupyText may be used to convert between 
```

Author: @chrisjsewell what do you use for spell check in `vscode` -- shamelessly going to copy your dev environment :-)

Reviewer 2: Haha, well I use [code-spell-checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker). But if you want to copy me completely then all my setting are synced to GitHub (with [settings-sync](https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync)):
https://gist.github.com/chrisjsewell/0e808e0f42c1a0b02d393df767027722 😄

Author: well that's awesome :-) I just came across this https://marketplace.visualstudio.com/itemdetails?itemName=Shan.code-settings-sync this week. Looks awesome.

ipynb and myst files in quasi-realtime to enable joint editing between the 
two formats. 

**Lead:** [Aakash](https://github.com/AakashGfude)

## Parsers

### [myst-parser](https://github.com/ExecutableBookProject/MyST-Parser) [Internal]

**Description:** 

An extended commonmark compliant parser, with bridges to docutils/sphinx

**Lead:** [Chris Sewell](https://github.com/chrisjsewell)

> For supported syntax please refer to [here](https://github.com/ExecutableBookProject/MyST-Parser#parsed-token-classes)

Reviewer: ```suggestion
For supported syntax please refer to [the myst-parser documentation](https://github.com/ExecutableBookProject/MyST-Parser#parsed-token-classes)
```

in general, I try to avoid hyper-linking words like "here" or "this". It makes the link harder to notice

> 

Reviewer 2: ```suggestion
*Phase I*

- [ ] Provide a parser that can convert raw text, written in an extended Markdown syntax (including roles and directives), to the `docutils` AST (used in Sphinx).

*Phase II*

Provide language support for user editing of documents written in MyST, in the form of:

- [ ] A reference static syntax highlighter.
- [ ] A standalone linting / diagnostic tool

*Phase III*

- [ ] Build on the tools developed in Phase II to introduce a full Language Support Protocol (LSP) implementation, including:
    - document tree stucture views
    - autocompletion for role and directive names and options.
    - hover views for dollar enclosed math
    - real-time linting diagnostics

```

### [myst-nb](https://github.com/ExecutableBookProject/MyST-NB) [Internal]

**Description:** Enables sphinx to read IPYNB files that contain
myst based syntax

> **Lead:** [Chris Holdgraf](https://github.com/choldgraf)

Reviewer: Just a note - I have been trying to get this to a workable "alpha" state but probably will shift my attention away from it and towards making proper connections with Jupyter Book HTML/CSS/JS...depending on whether @AakashGfude is able to help out either here or on the HTML piece. In general I guess my point is that this will definitely need more development support than just me.

Author: For sure. My main idea on `lead` as a coordinating point of contact -- not developer. Thanks for the update on this. In coming weeks I should get some time to look at `myst-nb` also.

Reviewer 3: @choldgraf @mmcky happy to help on the HTML/CSS/JS part.

Author: @choldgraf I have changed `lead` to `coordinator` as that was my initial intent.
